src/main/java/io/supertokens/inmemorydb/queries/PasswordlessQueries.java

```diff
@@ -0,0 +1,452 @@
+        String QUERY = "DELETE FROM " + Config.getConfig(start).getPasswordlessCodesTable() + " WHERE code_id = ?";
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            pst.setString(1, codeId);
+            pst.executeUpdate();
+        }
+    }
+
+    public static void deleteCodesOfDeviceBefore_Transaction(Start start, Connection con, String deviceIdHash,
+            long time) throws SQLException {
+        String QUERY = "DELETE FROM " + Config.getConfig(start).getPasswordlessCodesTable()
+                + " WHERE device_id_hash = ? AND created_at < ?";
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            pst.setString(1, deviceIdHash);
+            pst.setLong(2, time);
+            pst.executeUpdate();
+        }
+    }
+
+    public static void createUser_Transaction(Start start, Connection con, String userId, String email,
+            String phoneNumber, long timeJoined) throws SQLException {
+        {
+            String QUERY = "INSERT INTO " + Config.getConfig(start).getUsersTable()
+                    + "(user_id, recipe_id, time_joined)" + " VALUES(?, ?, ?)";
+            try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+                pst.setString(1, userId);
+                pst.setString(2, RECIPE_ID.PASSWORDLESS.toString());
+                pst.setLong(3, timeJoined);
+                pst.executeUpdate();
+            }
+        }
+
+        {
+            String QUERY = "INSERT INTO " + Config.getConfig(start).getPasswordlessUsersTable()
+                    + "(user_id, email, phone_number, time_joined)" + " VALUES(?, ?, ?, ?)";
+            try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+                pst.setString(1, userId);
+                pst.setString(2, email);
+                pst.setString(3, phoneNumber);
+                pst.setLong(4, timeJoined);
+                pst.executeUpdate();
+            }
+        }
+    }
+
+    public static void updateUser_Transaction(Start start, Connection con, String userId, String email,
+            String phoneNumber) throws SQLException {
+        String QUERY = "UPDATE " + Config.getConfig(start).getPasswordlessUsersTable()
+                + " SET email = ?, phone_number = ? WHERE user_id = ?";
+
+        try (PreparedStatement pst = con.prepareStatement(QUERY)) {
+            pst.setString(1, email);
+            pst.setString(2, phoneNumber);
+            pst.setString(3, userId);
+            pst.executeUpdate();
+        }
+    }
+
+    public static PasswordlessDevice getDevice(Start start, String deviceIdHash)
+            throws StorageQueryException, SQLException {
+        try (Connection con = ConnectionPool.getConnection(start)) {
+            return getDevice_Transaction(start, con, deviceIdHash);
+        }
+    }
+
+    public static PasswordlessDevice[] getDevicesByEmail(Start start, @Nonnull String email)
```

Comment:
It's split them like this because the `PasswordlessDevice` type in the core doesn't contain the codes since we don't always need them when getting a device. We could change this to be better for MongoDB, but earlier, we said we wouldn't consider that too much.